### ISSUE TITLE
Make twisted/python/test/test_fakepwd.py pass under Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5819]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)| @vperic reported|
|-|-|
|Trac ID|trac#5819|
|Type|task|
|Created|2012-07-28 12:01:17Z|
|Branch|https://github.com/twisted/twisted/tree/test_fakepwd-5819|

As before, to actually make this actually run you need to import unittest from the library, not from trial, and then "python3.2 -m unittest twisted.python.test.test_fakepwd" will run the tests. In this file, there is another issue - some tests are skipped (unless running as root), but the upstream unittest module has a different syntax for skipping tests than trial and reports these as a failure. They *do* work though: either run them as root to see, or trust me. :)

The change was the .next() method to the next() function in a couple of lines. I also used the opportunity to change the syntax of tests from the deprecated "assertEquals" to "asserEqual".

Attachments:

* [test_fakepwd.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cb2/cb2d4e3e631d46f659c133f19257f5a9d2494c54/920476f2bbcef87c1f22584609614745e9af2620.patch) (5136 bytes) - added by vperic on 2012-07-28 12:02:18Z - First version

<details><summary>Searchable metadata</summary>

```
trac-id__5819 5819
type__task task
reporter__vperic vperic
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_test_fakepwd_5819 branches/test_fakepwd-5819
branch_author__vperic vperic
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1343476877000000 1343476877000000
changetime__1343812542000000 1343812542000000
version__None None
owner__vperic vperic

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)<a name="note_1"></a>|@pitrou set owner to @vperic|
|-|-|

Looks good to me.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_2"></a>|@vperic commented|
|-|-|

(In [35015]) Create branch test_fakepwd-5819

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_3"></a>|@vperic commented|
|-|-|

[Buildbot run.](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/test_fakepwd-5819) I ordered just the changed test to be run.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_4"></a>|@vperic set status to `closed`|
|-|-|

(In [35017]) Merge test_fakepwd-5819: Make t.p.test.test_fakepwd pass under Python3

Author: vperic
Reviewer: antoine
Fixes: [#5819](https://github.com/twisted/twisted/issues/5819)

In two cases, the next() method of iterators was used; change this to
use the next() function to make the tests Python3 compatible.

Also use the opportunity to change the deprecated "assertEquals" syntax
to "assertEqual".
